plugins/acpi-wpbt/fu-plugin-acpi-wpbt.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-hash.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+void
+fu_plugin_add_security_attrs (FuPlugin *plugin, FuSecurityAttrs *attrs)
+{
+	g_autofree gchar *fn = NULL;
+	g_autofree gchar *path = NULL;
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+
+	/* create attr */
+	attr = fwupd_security_attr_new (FWUPD_SECURITY_ATTR_ID_SUSPEND_TO_IDLE);
+	fwupd_security_attr_set_plugin (attr, fu_plugin_get_name (plugin));
+	fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_RUNTIME_ISSUE);
+	fu_security_attrs_append (attrs, attr);
+
+	/* load WPBT table */
+	path = fu_common_get_path (FU_PATH_KIND_ACPI_TABLES);
+	fn = g_build_filename (path, "WPBT", NULL);
+	if (g_file_test (fn, G_FILE_TEST_EXISTS)) {
```

Comment:
Is testing it exists enough for this? I would think you want to make sure it's actually a windows binary?

Reply:
Have you got a link to the WPBT spec? I can sure enough check for the PE header.

Reply:
https://view.officeapps.live.com/op/view.aspx?src=http%3A%2F%2Fdownload.microsoft.com%2Fdownload%2F8%2FA%2F2%2F8A2FB72D-9B96-4E2D-A559-4A27CF905A80%2Fwindows-platform-binary-table.docx

Reply:
Looking at that spec the WPBT itself doesn't have the binary, but rather points to memory that will have the binary.  I don't anticipate that the memory region is actually accessible from Linux userland.  So I think that leaves two options of varying degrees of difficulty:
1. Parse the table from fwupd, make sure if it measures out to that spec.
2. Add a new kernel module for parsing the table and _exporting_ out the binary to userland in a sysfs interface.  I think this is most interesting for a researchers interface of being able to do some analysis on actual things with what is in WPBT and probably overkill.

If it looks strikingly like a duck and it's quacks at least checksum to that of a duck;'s it's probably a duck.

Reply:
I also agree it's not terribly useful to see the exe itself, unless of course we can verify the checksum isn't one of the known-bad ones -- but that's getting dangerously close to an _AV scanner_, and I'm not sure that's where I want this to head.

I've added a test to check the file is readable and to verify the checksum isn't invalid.